daemon/analytics.c

```diff
@@ -0,0 +1,874 @@
+    debug(D_ANALYTICS, "NETDATA_CONFIG_IS_PARENT           : [%s]", analytics_data.netdata_config_is_parent);
+    debug(D_ANALYTICS, "NETDATA_CONFIG_HOSTS_AVAILABLE     : [%s]", analytics_data.netdata_config_hosts_available);
+    debug(D_ANALYTICS, "NETDATA_HOST_CLOUD_AVAILABLE       : [%s]", analytics_data.netdata_host_cloud_available);
+    debug(D_ANALYTICS, "NETDATA_HOST_ACLK_AVAILABLE        : [%s]", analytics_data.netdata_host_aclk_available);
+    debug(D_ANALYTICS, "NETDATA_HOST_ACLK_IMPLEMENTATION   : [%s]", analytics_data.netdata_host_aclk_implementation);
+    debug(D_ANALYTICS, "NETDATA_HOST_AGENT_CLAIMED         : [%s]", analytics_data.netdata_host_agent_claimed);
+}
+
+/*
+ * Free data
+ */
+void analytics_free_data(void)
+{
+    freez(analytics_data.netdata_config_stream_enabled);
+    freez(analytics_data.netdata_config_memory_mode);
+    freez(analytics_data.netdata_config_exporting_enabled);
+    freez(analytics_data.netdata_exporting_connectors);
+    freez(analytics_data.netdata_allmetrics_prometheus_used);
+    freez(analytics_data.netdata_allmetrics_shell_used);
+    freez(analytics_data.netdata_allmetrics_json_used);
+    freez(analytics_data.netdata_dashboard_used);
+    freez(analytics_data.netdata_collectors);
+    freez(analytics_data.netdata_collectors_count);
+    freez(analytics_data.netdata_buildinfo);
+    freez(analytics_data.netdata_config_page_cache_size);
+    freez(analytics_data.netdata_config_multidb_disk_quota);
+    freez(analytics_data.netdata_config_https_enabled);
+    freez(analytics_data.netdata_config_web_enabled);
+    freez(analytics_data.netdata_config_release_channel);
+    freez(analytics_data.netdata_mirrored_host_count);
+    freez(analytics_data.netdata_mirrored_hosts_reachable);
+    freez(analytics_data.netdata_mirrored_hosts_unreachable);
+    freez(analytics_data.netdata_notification_methods);
+    freez(analytics_data.netdata_alarms_normal);
+    freez(analytics_data.netdata_alarms_warning);
+    freez(analytics_data.netdata_alarms_critical);
+    freez(analytics_data.netdata_charts_count);
+    freez(analytics_data.netdata_metrics_count);
+    freez(analytics_data.netdata_config_is_parent);
+    freez(analytics_data.netdata_config_hosts_available);
+    freez(analytics_data.netdata_host_cloud_available);
+    freez(analytics_data.netdata_host_aclk_available);
+    freez(analytics_data.netdata_host_aclk_implementation);
+    freez(analytics_data.netdata_host_agent_claimed);
+}
+
+/*
+ * Set a numeric/boolean data with a value
+ */
+void analytics_set_data(char **name, char *value)
+{
+    if (*name)
+        freez(*name);
+    *name = strdupz(value);
+}
+
+/*
+ * Set a string data with a value
+ */
+void analytics_set_data_str(char **name, char *value)
+{
+    size_t value_string_len;
+    if (*name)
+        freez(*name);
+    value_string_len = strlen(value) + 3;
+    *name = mallocz(sizeof(char) * value_string_len);
+    snprintfz(*name, value_string_len, "\"%s\"", value);
```

Comment:
In normal situation when a vector has a size `N`, we pass to functions like `sprintfz` a smaller size `N - 1` to keep code safe, are you sure that it is better to give as argument the same size here?

Reply:
Ok, modified a bit to provide `-1`.